x/evidence/keeper/infraction.go

```diff
@@ -26,19 +27,6 @@ func (k Keeper) HandleEquivocationEvidence(ctx sdk.Context, evidence *types.Equi
 	logger := k.Logger(ctx)
 	consAddr := evidence.GetConsensusAddress()
 
-	if _, err := k.slashingKeeper.GetPubkey(ctx, consAddr.Bytes()); err != nil {
```

Comment:
I might have asked this before, but is it okay that we remove this from the evidence module on the provider chain?

Reply:
Do you have any thoughts on this, @alexanderbez ?

Reply:
If public keys of validator's do not exist on the consumer chain, then this is OK I think.

Reply:
Wouldn't this module act as the evidence module on the provider chain?

Reply:
You're right. I can't see how it couldn't also act on the provider chain as well. We should address this.

Reply:
I added back the condition and updated it to verify only validators' public key on provider chains.  
Let's discuss if it's worth it.